src/pybind/wrapper.cpp

```diff
@@ -0,0 +1,221 @@
+        // x = ...
+        py::exec(R"(
+                from nmodl.ode import integrate2c
+                exception_message = ""
+                try:
+                    solution = integrate2c(equation_string, dt_var, vars,
+                                           use_pade_approx)
+                except Exception as e:
+                    # if we fail, fail silently and return empty string
+                    solution = ""
+                    exception_message = str(e)
+            )",
+                 py::globals(),
+                 locals);
+    } else {
+        // nothing to do, but the caller should know.
+        return;
+    }
+    solution = locals["solution"].cast<std::string>();
+    exception_message = locals["exception_message"].cast<std::string>();
+}
+
+void AnalyticDiffExecutor::operator()() {
+    auto locals = py::dict("expressions"_a = expressions, "vars"_a = used_names_in_block);
+    py::exec(R"(
+                            from nmodl.ode import differentiate2c
+                            exception_message = ""
+                            try:
+                                rhs = expressions[-1].split("=", 1)[1]
+                                solution = differentiate2c(rhs,
+                                                           "v",
+                                                           vars,
+                                                           expressions[:-1]
+                                           )
+                            except Exception as e:
+                                # if we fail, fail silently and return empty string
+                                solution = ""
+                                exception_message = str(e)
+                        )",
+             py::globals(),
+             locals);
+    solution = locals["solution"].cast<std::string>();
+    exception_message = locals["exception_message"].cast<std::string>();
+}
+
+SolveLinearSystemExecutor* create_sls_executor_func() {
+    return new SolveLinearSystemExecutor();
+}
+
+SolveNonLinearSystemExecutor* create_nsls_executor_func() {
+    return new SolveNonLinearSystemExecutor();
+}
+
+DiffeqSolverExecutor* create_des_executor_func() {
+    return new DiffeqSolverExecutor();
+}
+
+AnalyticDiffExecutor* create_ads_executor_func() {
+    return new AnalyticDiffExecutor();
+}
+
+void destroy_sls_executor_func(SolveLinearSystemExecutor* exec) {
+    delete exec;
+}
+
+void destroy_nsls_executor_func(SolveNonLinearSystemExecutor* exec) {
+    delete exec;
+}
+
+void destroy_des_executor_func(DiffeqSolverExecutor* exec) {
+    delete exec;
+}
+
+void destroy_ads_executor_func(AnalyticDiffExecutor* exec) {
+    delete exec;
+}
+
+void initialize_interpreter_func() {
+    pybind11::initialize_interpreter(true);
+    const auto python_path_cstr = std::getenv("PYTHONPATH");
+    if (python_path_cstr) {
+        pybind11::module::import("sys").attr("path").cast<pybind11::list>().insert(
```

Comment:
just curious ! if the `PYTHONPATH` is set before launching the nmodl, it doesn't get used? i.e. it needs to be inserted like this?

Reply:
That is correct. This because we are running an embedded python. It is up to the programmer to provide the environment to libpython. If you are running the python executable, it will do this for you.
This is a very interesting and relevant discussion on this topic: https://bugs.python.org/issue22213

Reply:
ok! thanks for the reference!